### PR TITLE
enforcing character limits

### PR DESCRIPTION
**Pull request by ethanmichel0**
Adressing what Deena said during stand up about how often attributions and sub attributions are too long.

### CODE REVIEW

File: src/CommunityVoices/Public/js/quote-form.js [hunk 1]
```diff
@@ -12,3 +12,14 @@ function clickEdited() {
     originalText.value = '';
   }
 }
+
+function displayCount(box,count) {
+  var countDisplay = document.getElementById(count);
```
Comment (Sammidysam) [inline src/CommunityVoices/Public/js/quote-form.js:17]: Please use `const` here.  I can explain why sometime if you'd like, but `var` is somewhat evil.

File: src/CommunityVoices/Public/js/quote-form.js [hunk 2]
```diff
@@ -12,3 +12,14 @@ function clickEdited() {
...
+  if(box.hasAttribute("suggestedlength") && (box.value.length > box.getAttribute("suggestedlength"))) {
+    countDisplay.innerHTML = box.value.length + " / " + box.getAttribute("maxlength") + " characters -- Please shorten!";
+    countDisplay.style.color = "red";
+  } else {
+    countDisplay.innerHTML = box.value.length + " / " + box.getAttribute("maxlength") + " characters";
```
Comment (Sammidysam) [inline src/CommunityVoices/Public/js/quote-form.js:22]: This line is repetitive.  I think that the `countDisplay.innerHTML` should be set to this line regardless and the addendum for suggested length should be added to it in a conditional.

Comment (Sammidysam) [reply src/CommunityVoices/Public/js/quote-form.js:22]: Always try to make your conditionals affect only the code that is truly different between the two.

**Comment by ethanmichel0** [comment]
> This is great, but I have a question (maybe this is for @rehannan ):
> Do we seek to _actually_ set a max length to attributions and subattributions, or is it only a suggestion? As it stands, everything written in JavaScript is always a suggestion - someone could always make a request from their command prompt that sends an attribution that is longer than the length that JavaScript enforces. If we truly want to set a max length for these fields, we need to do it inside of PHP.

How would I enforce this in PHP? Do you mean a SQL restriction/rule? I have this as an xslt attribute right now, which is presented on the website. So wouldn't anyone interfacing with the website need to go through the xslt input box in order to submit attributes/subattributes?

**Comment by Sammidysam** [comment]
Why did you close this?

**Comment by Sammidysam** [comment]
I feel like I have not been able to clearly express this well.  The outside world can interact with the API layer.  For example, see https://www.environmentaldashboard.org/community-voices/api/quotes
Furthermore, I can even submit a quote through the command line if I want even.
```
curl -d "text=thing" -X POST http://localhost:3001/community-voices/api/quotes
```
It won't work because I am missing enough data for it to work, but I am just trying to say that I do not have to use `environmentaldashboard.org` to interact with the website.

**Comment by Sammidysam** [comment]
SQL restriction or `validateForUpload` can be used to truly enforce character limits, yes.

**Comment by Sammidysam** [comment]
Maybe Mozilla is a better teacher than me.  See third paragraph of https://developer.mozilla.org/en-US/docs/Learn/Forms/Form_validation and possibly follow their web security link.

File: src/CommunityVoices/Public/js/quote-form.js (hunk 1 above)
Comment (ethanmichel0) [reply src/CommunityVoices/Public/js/quote-form.js]: I addressed your JS comments in new commit.

**Comment by Sammidysam** [comment]
Nice.  I think displaying that dash as ` - ` rather than `--` may be more consistent to other places across our applications, but I don't think this is a big deal at all.

**Pull request merged**